### PR TITLE
Don't build unless the source or build-config changed

### PR DESCRIPTION
**Pull request by victorporof**
Fixes https://github.com/mozilla/tofino/issues/684

@Mossop

### CODE REVIEW

**Comment by victorporof** [comment]
Appveyor failure looks unrelated https://ci.appveyor.com/project/Mozilla/tofino-u1hv8/build/1.0.558-rognxmcj/job/8gtn9nrb4hi2l70u

File: build/utils.js [hunk 1]
```diff
@@ -241,6 +242,40 @@ export function transpile(filename, options = {}) {
   });
 }
 
+export function digest(source, alg = 'sha1') {
+  return new Promise((resolve, reject) => {
+    dirsum.digest(source, alg, (err, results) => {
```
Comment (Mossop) [inline build/utils.js]: Can't we just use thenify here?

File: build/task-build-browser.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Any copyright is dedicated to the Public Domain.
 // http://creativecommons.org/publicdomain/zero/1.0/
 
+import 'colors';
```
Comment (Mossop) [inline build/task-build-browser.js]: Use colors-safe.

**Comment by Mossop** [comment]
LGTM

**Pull request merged**